cas-server-support-ldap/src/main/java/org/jasig/cas/adaptors/ldap/services/LdapServiceRegistryDao.java

```diff
@@ -114,39 +115,43 @@ public RegisteredService save(final RegisteredService rs) {
      * @return the registered service
      */
     private RegisteredService update(final RegisteredService rs) {
+        String currentDn = null;
         Connection searchConnection = null;
         try {
             searchConnection = getConnection();
             final Response<SearchResult> response = searchForServiceById(searchConnection, rs.getId());
             if (hasResults(response)) {
-                final String currentDn = response.getResult().getEntry().getDn();
-
-                Connection modifyConnection = null;

```

Comment:
Because the searchConnection has to be closed before we do anything else. The check is of course added below once the connection is cleaned up.